src/python/WMCore/Services/WMStats/WMStatsReader.py

```diff
@@ -370,26 +378,36 @@ def isWorkflowCompletedWithLogCollectAndCleanUp(self, requestName):
         reqInfoInstance = RequestInfo(requestInfo[requestName])
         return reqInfoInstance.isWorkflowFinished()
 
-    def getTaskJobSummaryByRequest(self, requestName, sampleSize=1):
-
+    def getTaskJobSummaryByRequest(self, requestName, sampleSize=1, usePycurl=True):
+        reqStart = time.time()
         options = {'reduce': True, 'group_level': 5, 'startkey': [requestName],
                    'endkey': [requestName, {}]}
         results = self._getCouchView("jobsByStatusWorkflow", options)
+
         jobDetails = {}
-        for row in results['rows']:
-            # row["key"] = ['workflow', 'task', 'jobstatus', 'exitCode', 'site']
-            startKey = row["key"][:4]
-            endKey = []
-            site = row["key"][4]
-            if site:
-                startKey.append(site)
-
-            endKey.extend(startKey)
-            endKey.append({})
-            numOfError = row["value"]
-
-            jobInfo = self.jobDetailByTasks(startKey, endKey, numOfError, sampleSize)
+        if usePycurl is True:
+            jobInfo = getTaskJobSummaryByRequestPycurl(results, sampleSize, self.serviceOpts)
             jobDetails = nestedDictUpdate(jobDetails, jobInfo)
+        else:
+            # then it is sequential
+            for row in results['rows']:
```

Comment:
If we'll consider converting `jobDetailByTasksPycurl` to a generator then code over here should be re-factored as well to be a generator. To make such transition easier the code below first should be put into individual function and then we can convert this function to a generator with common data-structure.